pkg/sync/sync_context.go

```diff
@@ -810,13 +811,27 @@ func (sc *syncContext) autoCreateNamespace(tasks syncTasks) syncTasks {
 					tasks = append(tasks, nsTask)
```

Comment:
I am not sure if this is still correct with the new `namespaceModifier` logic. It seems that it is appending the `nsTask` without potential labels/annotations. Shouldn't we always invoke the `sc.namespaceModifier()` in case of namespace creation?